esx_service/vmdk_ops.py

```diff
@@ -314,15 +314,25 @@ def find_child(vm_name):
 
 #returns error, or None for OK
 def attachVMDK(vmdk_path, vm_name):
-    vm = findVmByName(vm_name)
+    try:
+       vm = findVmByName(vm_name)
+    except Exception as e:
+       msg = "VM {0} not found".format(vm_name)
```

Comment:
We have no idea what exceptions ESX is going to return save VimFault which
is a catch all and how does it help  catching a VimFault and ignoring it.

On Thu, Jun 2, 2016 at 12:47 PM, Mark Sterin <notifications@github.com>
wrote:

> In esx_service/vmdk_ops.py
> <https://github.com/vmware/docker-volume-vsphere/pull/433#discussion_r65493899>
> :
>
> > @@ -314,15 +314,25 @@ def find_child(vm_name):
> >
> >  #returns error, or None for OK
> >  def attachVMDK(vmdk_path, vm_name):
> > -    vm = findVmByName(vm_name)
> > +    try:
> > +       vm = findVmByName(vm_name)
> > +    except Exception as e:
> > +       msg = "VM {0} not found".format(vm_name)
>
> catching generic exception is frown upon in python. We should either stick
> with C-style error code unrolling, or use specific exceptions
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/vmware/docker-volume-vsphere/pull/433/files/1ae935fbbc0c3205389f8fd83888b9afea83cba2#r65493899>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/APHseGmb16gxUtl8dEs_iJT2al40OG9rks5qHoOBgaJpZM4IrROI>
> .
>